urequests/urequests.py

```diff
@@ -39,7 +39,13 @@ def request(method, url, data=None, json=None, headers={}, stream=None):
     if proto != "http:":
         raise ValueError("Unsupported protocol: " + proto)
 
-    ai = usocket.getaddrinfo(host, 80)
+    port = 80
+    if ':' in host:
+        _host, port = host.split(':', 1)
+        host = _host
```

Comment:
This is still superfluous. MicroPython allocates a memory slot for each local variable, more variables == more memory used. So, if it's possible to reuse a variable, that should be done - on the level of micropython-lib for sure. I'm fixing this myself.